docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -1069,6 +1069,46 @@ public class NoopAsyncObserverExceptionHandler implements AsyncObserverException
 }
 ----
 
+[[reactive_pitfalls]]
+== CDI Pitfalls with Reactive Programming
+
+CDI is a purely synchronous framework.
+Its notion of asynchrony is very limited and based solely on thread pools and thread offloading.
+Therefore, there is a number of pitfalls when using CDI together with reactive programming.
+
+=== Detecting When Blocking Is Allowed
+
+The `BlockingOperationControl.isBlockingAllowed()` method allows detecting when blocking is allowed.
+When it is not, and you need to perform a blocking operation, you have to offload it to another thread.
```

Comment:
Maybe we should give an example of a blocking operation or add a link to Quarkus/external docs?

Reply:
Maybe best to do that in the source code listing? Say, just `Thread.sleep(5000);`?

Reply:
Yes, maybe with a comment like:
```java
// Write data to the filesystem, use JDBC, call a REST service, etc.
Thread.sleep(5000);
```

Reply:
Done.